lib/repl.js

```diff
@@ -1136,15 +1125,14 @@ function complete(line, callback) {
                 }
               }
             } catch (e) {
-              //console.log("completion error walking prototype chain:" + e);
+              // console.log("completion error walking prototype chain:" + e);
             }
           }
 
           if (memberGroups.length) {
             for (i = 0; i < memberGroups.length; i++) {
-              completionGroups.push(memberGroups[i].map(function(member) {
-                return expr + '.' + member;
-              }));
+              completionGroups.push(
+                memberGroups[i].map((member) => expr + '.' + member));
```

Comment:
const foo = memberGroups.reduce(( acc, memberGroup ) => {
    acc.push(memberGroup.map((member) => `${ expr }.${ filter }`));
    return acc;
}, []);
completionGroups.concat(foo);
Don't mind the name foo as could not think of some descent name but this will make the code immutable. Although we are mutating completionGroups array but in next change we can convert it also to immutability

Reply:
That is not necessary out of my perspective. I personally feel the current code is more expressive than using `reduce`.